model/driver/src/icon4py/model/driver/icon4py_driver.py

```diff
@@ -260,13 +247,43 @@ def _do_dyn_substepping(
             do_clean_mflx = False
 
             if not self._is_last_substep(dyn_substep):
-                self._swap()
+                prognostic_state_swp.swap()
 
             self._is_first_step_in_simulation = False
 
         # TODO (Chia Rui): compute airmass for prognostic_state here
 
 
+class DriverStates(NamedTuple):
```

Comment:
I don't know whether that makes sense, as I don't know wether all these state `structs`make sense. Potentially for every model component `X` you will have a different `XDiagnosticState` because there are so many diagnostic variables and each component uses a particular subset... 

But we might just introduce it for now and maybe later skip it. It makes the `initialize` function a bit easier to grasp. That said, I don't like the `initialize`at all. It does too many things at the time. I would split it apart.